### PR TITLE
neonlabsorg/Solana-EVM#355 Reduce logging on Indexer

### PR DESCRIPTION
**Pull request by otselnik**
Introduced `INDEXER_LOG_SKIP_COUNT` environment variable, by default set to 10.
Introduced `CountedLogger` to print log once in `INDEXER_LOG_SKIP_COUNT` calls.
Combined multiple lines of output into one line.
Removed useless logs.

### CODE REVIEW

File: proxy/indexer/indexer.py [hunk 1]
```diff
@@ -238,14 +238,19 @@ def complete_done_txs(self):
         for tx in self._done_tx_list:
             self.unmark_ix_used(tx)
             if tx.neon_tx.is_valid() and tx.neon_res.is_valid():
-                with logging_context(req_id=tx.neon_tx.sign[:7]):
+                with logging_context(neon_id=tx.neon_tx.sign[:7]):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/indexer.py]: `neon_trx_id`

File: proxy/environment.py [hunk 2]
```diff
@@ -44,6 +44,7 @@
 GET_SOL_PRICE_RETRY_INTERVAL = int(os.environ.get("GET_SOL_PRICE_RETRY_INTERVAL", 1))
 GET_WHITE_LIST_BALANCE_MAX_RETRIES = int(os.environ.get("GET_WHITE_LIST_BALANCE_MAX_RETRIES", 3))
 GET_WHITE_LIST_BALANCE_RETRY_INTERVAL_S = int(os.environ.get("GET_WHITE_LIST_BALANCE_RETRY_INTERVAL_S", 1))
+INDEXER_LOG_SKIP_COUNT = int(os.environ.get("INDEXER_LOG_SKIP_COUNT", 10))
```
Comment (rozhkovdmitrii) [inline proxy/environment.py:47]: Does it really need to be an environment. 

```python
class CountedLogger:
    INDEXER_LOG_SKIP_COUNT = 10
...
        if self.counter % self.INDEXER_LOG_SKIP_COUNT == 0:
...
```

File: proxy/indexer/indexer.py [hunk 3]
```diff
@@ -666,6 +671,57 @@ def execute(self) -> bool:
         return self._decode_tx(tx)
 
 
+@logged_group("neon.Indexer")
+class BlocksIndexer:
+    def __init__(self, db: IndexerDB, client):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/indexer.py]: `solana_client: Client` or at least annotation

File: proxy/indexer/utils.py [hunk 4]
```diff
@@ -121,6 +124,36 @@ def get_code_from_account(client: Client, address, *, logger):
     return '0x' + data[offset:][:storage.code_size].hex()
 
 
+class CountedLogger:
+    def __init__(self):
+        self.set_defaults()
+
+    def set_defaults(self):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]: `def reset(self):`

File: proxy/indexer/utils.py [hunk 5]
```diff
@@ -121,6 +124,36 @@ def get_code_from_account(client: Client, address, *, logger):
...
+        self.counter = 0
+        self.items_list = {}
+        self.items_latest = {}
+
+    def print(self, logger, list_params, latest_params):
+        for key, value in list_params.items():
+            if key not in self.items_list:
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]:
```
            metric_list = self.items_list.set_default(key, [])
            metric_list.append(value)
```

File: proxy/indexer/utils.py [hunk 6]
```diff
@@ -121,6 +124,36 @@ def get_code_from_account(client: Client, address, *, logger):
...
+class CountedLogger:
+    def __init__(self):
+        self.set_defaults()
+
+    def set_defaults(self):
+        self.counter = 0
+        self.items_list = {}
+        self.items_latest = {}
+
+    def print(self, logger, list_params, latest_params):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]: Annotations

File: proxy/indexer/utils.py [hunk 7]
```diff
@@ -121,6 +124,36 @@ def get_code_from_account(client: Client, address, *, logger):
...
+        for key, value in list_params.items():
+            if key not in self.items_list:
+                self.items_list[key] = []
+            self.items_list[key].append(value)
+        for key, value in latest_params.items():
+            self.items_latest[key] = value
+        self.counter += 1
+
+        if self.counter % INDEXER_LOG_SKIP_COUNT == 0:
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]:
```
if self. ... != 0:
    return
```

**Comment by rozhkovdmitrii** [comment]
image

File: proxy/indexer/indexer.py [hunk 8]
```diff
@@ -751,7 +804,7 @@ def process_receipts(self):
 
             for _ in ix_info.iter_ixs():
                 req_id = ix_info.sign.get_req_id()
-                with logging_context(req_id=req_id):
+                with logging_context(sol_id=req_id):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/indexer.py]: sol_trx_id

File: proxy/indexer/utils.py [hunk 9]
```diff
@@ -121,6 +124,36 @@ def get_code_from_account(client: Client, address, *, logger):
     return '0x' + data[offset:][:storage.code_size].hex()
 
 
+class CountedLogger:
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]: MetricsToLogBuff ?

File: proxy/indexer/indexer.py (hunk 1 above)
Comment (otselnik) [reply proxy/indexer/indexer.py]: This name is pretty obvious, and I don't want to make it too long.

File: proxy/environment.py (hunk 2 above)
Comment (otselnik) [reply proxy/environment.py:47]: By default, it does not need to be set on the environment, but if somebody would want statistics every turn, he can set this variable.

Comment (rozhkovdmitrii) [reply proxy/environment.py:47]: You're right to some extend but I don't see we have to pump up the configuration until it's actually needed.
If it needed then we will extend configuration. And I hope this idea is reasonable for both of us, we don't optimize anything before it's really needed.

File: proxy/indexer/utils.py [hunk 10]
```diff
@@ -121,6 +124,37 @@ def get_code_from_account(client: Client, address, *, logger):
...
+class MetricsToLogBuff :
+    def __init__(self):
+        self._reset()
+
+    def _reset(self):
+        self.counter = 0
+        self.items_list = {}
+        self.items_latest = {}
+
+    def print(self, logger: function, list_params: dict, latest_params: dict):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]:
```
    def print(self, logger: Callable[[str], None], list_params: Dict[str, float], latest_params: Dict[str, int]):
```
something like that

**Comment by rozhkovdmitrii** [comment]
> image

It can't be a proper logging note

File: proxy/indexer/utils.py [hunk 11]
```diff
@@ -124,34 +124,35 @@ def get_code_from_account(client: Client, address, *, logger):
...
+        self._reset()
 
-    def set_defaults(self):
+    def _reset(self):
         self.counter = 0
         self.items_list = {}
         self.items_latest = {}
 
-    def print(self, logger, list_params, latest_params):
+    def print(self, logger: function, list_params: dict, latest_params: dict):
```
Comment (rozhkovdmitrii) [inline proxy/indexer/utils.py]: from typing import Dict, Callable
 print(self, logger: Callable[...], list_params: Dict[...], latest_params: Dict[...]):

File: proxy/indexer/indexer.py (hunk 1 above)
Comment (afalaleev) [reply proxy/indexer/indexer.py]: neon_tx has the same length, but gives more information than neon_id

File: proxy/indexer/indexer.py (hunk 8 above)
Comment (afalaleev) [reply proxy/indexer/indexer.py]: sol_tx has the same length

**Review by rozhkovdmitrii — DISMISSED** [review]

**Review by afalaleev — APPROVED** [review]

**Review by rozhkovdmitrii — APPROVED** [review]

**Pull request merged**